### PR TITLE
Feature: Adminwallet

### LINKED ISSUE #30: Admin wallet proxy contract
1. contract that holds fuse tokens to top user wallets
2. contract that has whitelisting permissions
3. implements a "Role" that gives access admin access to the contract and use the above 2 methods
4. during deploy we give this contract the whitelisting permission
5. contract reimburse gas to caller, meaning it will top the account of the Admin address that called any of topwallet,whitelist actions
6. contracts initialize admins from constructor, constructor topwallet of admins

We will use this contract as a proxy for admin actions of whitelisting+topping wallet

API:
constructor(addresses[] admins)
addAdmins(addresses[]) only Owner
removeAdmins(addresses[]) only Owner
whitelist(address) only Admin
removeWhitelist(address) only Admin
blacklist(address) only Admin
removeBlacklist(address) only Admin
topWallet(address) only Admin
  - topWallet fills address balance to "X" Million GWEI
  - address can get filled only if balance <= 0.25*X
  - address can get filled max Y times a day

### PR DESCRIPTION
**Pull request by Maasdamind**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes #30  (link your issue here)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

# Checklist:
- [x] PR title matches follow: (Feature|Bug|Chore) Task Name
- [x] My code follows the style guidelines of this project
- [x] I have followed all the instructions described in the initial task (check Definitions of Done)
- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have added reference to a related issue in the repository
- [ ] I have added a detailed description of the changes proposed in the pull request. I am as descriptive as possible, assisting reviewers as much as possible.
- [ ] I have added screenshots related to my pull request ( for frontend tasks)
- [ ] I have pasted a gif showing the feature.
- [x] @mentions of the person or team responsible for reviewing proposed changes

### CODE REVIEW

**Review by sirpy — CHANGES_REQUESTED** [review]
1. Admin wallet needs to payback gas to caller
2. Top wallet should check target balance and reduce it from topping amount. Top only if topping amount greater than 0.5 of base topping account
3. Re-Verify you confirm to the story description and api

**Comment by sirpy** [comment]
Topwallet should transfer native eth not gd

**Comment by Maasdamind** [comment]
@sirpy The PR now conforms to the story description and has the described API calls

File: contracts/wallet/AdminWallet.sol [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+    /* @dev Modifier that checks if caller is admin of wallet
+     */
+    modifier onlyAdmin() {
+        require(isAdmin(msg.sender), "Caller is not admin");
+        _;
+    }
+
+    modifier reimburseGas() {
+        _;
+        require(msg.sender.balance <= toppingAmount.div(4), "Admin balance too high");
```
Comment (sirpy) [inline contracts/wallet/AdminWallet.sol]: This is going to generate exception and prevent normal operation if wallet doesn't require topping

**Review by sirpy — CHANGES_REQUESTED** [review]

**Pull request merged**